go/pkg/client/cas.go

```diff
@@ -438,6 +438,11 @@ func (c *Client) uploadNonUnified(ctx context.Context, data ...*uploadinfo.Entry
 					if err != nil {
 						return errors.Wrapf(err, "error while fetching FullData for digest %v", dg)
 					}
+
+					if dg.Size != int64(len(data)) {
+						return errors.Errorf("blob size changed while uploading, given:%d now:%d for %s", dg.Size, int64(len(data)), ue.Path)
```

Comment:
This looks like something that must never happen, so I'd be OK with even panic (like assertion), but this is good too

Reply:
Ack, let me return error this time.